common/src/main/kotlin/io/homeassistant/companion/android/common/data/HttpExceptionCallAdapter.kt

```diff
@@ -0,0 +1,63 @@
+package io.homeassistant.companion.android.common.data
+
+import io.homeassistant.companion.android.common.exception.HttpException
+import okhttp3.Request
+import okio.Timeout
+import retrofit2.Call
+import retrofit2.CallAdapter
+import retrofit2.Callback
+import retrofit2.Response
+import retrofit2.Retrofit
+import java.lang.reflect.ParameterizedType
+import java.lang.reflect.Type
+
+class HttpExceptionCallAdapterFactory : CallAdapter.Factory() {
```

Comment:
Added this adapter to minimize the amount of code changes. IMO we should replace it with an `Result` style adapter to remove exception-based error handling. Im happy to do it in this /separate PR.

Reply:
Changing every API to return a Result<> is a big task. I invite you to first create an issue with proper explanations. It needs to be in another PR (maybe multiples one), then adding a lint rule to force the usage of `Result`.

Currently exception handling in the application is not the best and would need huge changes in general (stopping using `catch(e:Exception)` that is way too broad for instance)